### ISSUE TITLE
[css-align] Allowing fallback alignments without breaking shorthands

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#1002]
When adding the `place-*` shorthands, we realized that specifying a fallback alignment for the content distribution keywords (`space-around` etc.) resulted in parsing ambiguities. We debating using a slash in the shorthand, but decided not to because it would make 2-axis alignment syntax inconsistent with similar syntax elsewhere in CSS, e.g. for `scroll-snap-align` or `background-position`.

An alternate idea is to use the slash to separate the fallback from the initial. Imho this also helps make it clearer that we're declaring a fallback alignment, since slashes are used to indicate alternates in general typographic usage.

Examples:

```
justify-content: space-around / center; /* space-around, fall back to center - both axes */
place-content: space-around / center; /* space-around both axes, fall back to center */
place-content: space-around start / center; /* space-around b-axis, start x-axis, fallback center */
```

(I think it's probably best to consider this for Level 2, leaving fallback keywords out for Level 1 so we can take Level 1 to CR asap.)

**Comment by tabatkins** [comment]
This sounds like the values are basically parallel; why not just use commas to separate the fallback from the normal?

**Comment by fantasai** [comment]
Resolution to defer to L4: https://lists.w3.org/Archives/Public/www-style/2017Apr/0027.html

**Comment by javifernandez** [comment]
i agree that a slash would be better used as a separator for row-axis and column-axis values. It even makes the shorthand's syntax clearer, IMO

**Comment by MatsPalmgren** [comment]
I don't think this syntax issue should be deferred to L4, because adopting the proposal to use `/` as the separator will be impossible at that point since it wouldn't be backwards compatible.  `place-content:space-between start`  currently means `align-content: space-between; justify-content: start` whereas if we adopt `/` as the separator and allow for fallback values in the shorthands, then it's a single value and means `align-content:space-between start; justify-content:space-between start`.

It seems to me that `/` was actually the preferred separator by most people when the shorthands were discussed in issue #595.  Using space as the separator was only accepted under the assumption that fallback values and `<overflow-position>` would never be allowed in the shorthands. It seems that assumption is now false and I would like to withdraw my "commenter satisfied" for issue #595, fwiw :-).

If you don't want to resolve on an extended syntax here for L3, then I request that these shorthands be deferred wholesale to L4 to avoid creating a situation were the preferred syntax can't be introduced because it would break the web.

**Comment by fantasai** [comment]
No, I was suggesting that the slash separates the initial and the fallback, not the axes. So we'd have `align-content: space-between / center` for the longhand. This would help make it clearer that `center` is a secondary value, vs. the current syntax `align-content: space-between center`.

Then for the shorthand, `place-content: space-between / center` would setting both `align-content` and `justify-content` to `space-between / center`. In cases where you want different values in each axis, we'd follow the `border-radius` shorthand pattern, e.g. `place-content: space-evenly / start center` would assign `align-content: space-evenly / start; justify-content: space-evenly / center`.

**Comment by fantasai** [comment]
(Syntax of `border-radius: [ <length> | <percentage> ]{1,4} [ / [ <length> | <percentage> ]{1,4} ]?`, described in https://www.w3.org/TR/css3-background/#corners, for reference.)

**Comment by javifernandez** [comment]
This issue still needs the changes in the spec. I know that Firefox started to implement the new behavior:

https://bugzilla.mozilla.org/show_bug.cgi?id=1430622

I'll start ASAP to implement it in WebKit and Blink as well, but I'd like to have the spec updated first. 
Thanks.

**Comment by emilio** [comment]
We haven't implemented the new thing yet (that's a compat fix, we weren't implementing `place-content` in the new style engine correctly).

But we can do so as soon as the spec is changed as well, I'll file and implement as soon as @MatsPalmgren gives me the ok :).

**Comment by MatsPalmgren** [comment]
FYI, we're waiting for the shorthand value grammar updates before implementing this.

**Comment by javifernandez** [comment]
We are waiting too before implementing it in WebKit and Blink.

**Comment by MatsPalmgren** [comment]
Adding to agenda to be discussed together with #5088.
